src/celeritas/user/StepData.hh

```diff
@@ -297,6 +297,66 @@ struct StepStateData
         step_length = other.step_length;
         particle = other.particle;
         energy_deposition = other.energy_deposition;
+        return *this;
+    }
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Gathered data and persistent scratch space for gathering and copying data.
+ *
+ * Extra storage \c scratch and \c valid_id is needed to efficiently gather and
+ * copy the step data on the device but will not be allocated on the host.
+ */
+template<Ownership W, MemSpace M>
+struct StepStateData
+{
+    //// TYPES ////
+
+    using StepSelectionData = StepSelectionStateData<W, M>;
+    template<class T>
+    using StateItems = celeritas::StateCollection<T, W, M>;
+
+    //// DATA ////
+
+    //! Gathered data for a single step
+    StepSelectionData step;
```

Comment:
Maybe `state` or `data`? Seems like we mostly use this as `step.step.{...}`, though I'm not sure `step.data` is too much an improvement...

Reply:
Eh, it's better than `step.step` at least. :)

Reply:
`step.data.step.step.data.data.step`